### PR TITLE
File input field component from UI Kit v5

### LINKED ISSUE #1780: File input field component customised from Boosted library, added to Forms category.
## Description
Update file input form control to comply with new design based on UI Kit v5.
Some minor modifications have been added to fil input.

image

### PR DESCRIPTION
**Pull request by hannahiss**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Related issues

Fixes #1780 

### Description

To teh left: design asked (from Zeplin), to the right: what I cando
image

### Motivation & Context

After update of ODS with new UI kit v5, we have to make some few changes on designs in Boosted.

### Types of change

- Bug fix (non-breaking which fixes an issues)
- New feature (non-breaking change which adds functionality)
- Refactoring (non-breaking change)
- Breaking change (fix or feature that would change existing functionality)

### Live previews

- <https://deploy-preview-1783--boosted.netlify.app/docs/5.2/forms/form-control/#file-input>

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

#### Accessibility

- [x] My change follows accessibility good practices; I have at least run axe

#### Design

- [ ] My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
- [x] My change is compatible with responsive display

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
- [ ] I have added JavaScript unit tests to cover my changes

#### Documentation

- (NA) My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- (NA) My change introduces changes to the migration guide
- (NA) My new component is added in Storybook
- (NA) My new component is compatible with RTL
- [ ] Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
- [x] Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
- [ ] Code review
- [ ] Design review
- [ ] A11y review

#### After the merge

- [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

**Comment by hannahiss** [comment]
## Tests with BrowserStack

### Windows/Chrome
image

### Windows/Firefox
image

### Mac/Safari 
image

### Window/Edge
image

### Android/Chrome
image

File: scss/_variables.scss [hunk 1]
```diff
@@ -1246,9 +1246,11 @@ $form-range-thumb-transition:              background-color $transition-duration
 // scss-docs-end form-range-variables
 
 // scss-docs-start form-file-variables
-$form-file-button-color:          $input-color !default;
-$form-file-button-bg:             var(--#{$prefix}body-bg) !default;
-$form-file-button-hover-bg:       var(--#{$prefix}secondary-bg) !default;
+$form-file-button-color:            $input-color !default;
+$form-file-placeholder-color:       $gray-700 !default; // Boosted mod
+$form-file-focus-border-color:      $black !default; // Boosted mod
```
Comment (louismaximepiton) [inline scss/_variables.scss:1251]:
```suggestion
$form-file-focus-border-color:      var(--#{$prefix}body-color) !default; // Boosted mod
```

or maybe just 
```suggestion
$form-file-focus-border-color:      $border-color !default; // Boosted mod
```

File: scss/_variables.scss [hunk 2]
```diff
@@ -1246,9 +1246,11 @@ $form-range-thumb-transition:              background-color $transition-duration
 // scss-docs-end form-range-variables
 
 // scss-docs-start form-file-variables
-$form-file-button-color:          $input-color !default;
-$form-file-button-bg:             var(--#{$prefix}body-bg) !default;
-$form-file-button-hover-bg:       var(--#{$prefix}secondary-bg) !default;
+$form-file-button-color:            $input-color !default;
+$form-file-placeholder-color:       $gray-700 !default; // Boosted mod
```
Comment (louismaximepiton) [inline scss/_variables.scss:1250]:
```suggestion
$form-file-placeholder-color:       var(--#{$prefix}secondary-color) !default; // Boosted mod
```

File: scss/forms/_form-control.scss [hunk 3]
```diff
@@ -24,10 +24,17 @@
 
   &[type="file"] {
     overflow: hidden; // prevent pseudo element button overlap
+    color: $form-file-placeholder-color; // color of the text in the right part of input
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:27]: Missing Boosted mod

File: scss/forms/_form-control.scss [hunk 4]
```diff
@@ -24,10 +24,17 @@
...
 
     &:not(:disabled):not([readonly]) {
       cursor: pointer;
     }
+
+    &:active:not(:disabled),
+    &:focus {
+      color: $form-file-placeholder-color;
+      border-color: $form-file-focus-border-color !important; // stylelint-disable-line declaration-no-important
+    }
```
Comment (louismaximepiton) [inline scss/forms/_form-control.scss:37]: Missing Boosted mod around

File: scss/_variables.scss [hunk 5]
```diff
@@ -1246,9 +1246,11 @@ $form-range-thumb-transition:              background-color $transition-duration
 // scss-docs-end form-range-variables
 
 // scss-docs-start form-file-variables
-$form-file-button-color:          $input-color !default;
-$form-file-button-bg:             var(--#{$prefix}body-bg) !default;
-$form-file-button-hover-bg:       var(--#{$prefix}secondary-bg) !default;
+$form-file-button-color:            $input-color !default;
```
Comment (louismaximepiton) [inline scss/_variables.scss:1249]: Any specific reason to add a tabulation here ?

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
Mainly missing Boosted mod and modifications due to the dark mode variables to use

FYI, I tried `:placeholder-shown` but didn't seem to work too. Just for future maintainers, can you consign the different solutions we didn't want to/couldn't do ? (js, force required, etc ..)

**Comment by Franco-Riccitelli** [comment]
I can confirm we agreed the File Input font colour should be always black, except for the disabled state. I have now made this change to the Web UI Kit for Sketch and will be published in the next release of the Kit.

**Comment by hannahiss** [comment]
After design review, and technical impossibility to realize fully the design that was wanted, Franco agreed to change it slightly to match the current design => no change is needed anymore

**Pull request closed without merging**